File: 1.1/src/cz/mg/vulkan/PFNvkGetQueryPoolResults.java
package cz.mg.vulkan;

public class PFNvkGetQueryPoolResults extends VkFunctionPointer {
    public PFNvkGetQueryPoolResults() {
    }

    protected PFNvkGetQueryPoolResults(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetQueryPoolResults(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetQueryPoolResults(long value) {
        setValue(value);
    }

    public PFNvkGetQueryPoolResults(VkInstance instance) {
        super(instance, new VkString("vkGetQueryPoolResults"));
    }

    public int call(VkDevice device, VkQueryPool queryPool, int firstQuery, int queryCount, long dataSize, VkObject pData, long stride, int flags){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), queryPool != null ? queryPool.getVkAddress() : VkPointer.getNullAddressNative(), firstQuery, queryCount, dataSize, pData != null ? pData.getVkAddress() : VkPointer.NULL, stride, flags);
    }

    protected static native int callNative(long vkaddress, long device, long queryPool, int firstQuery, int queryCount, long dataSize, long pData, long stride, int flags);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetQueryPoolResults.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetQueryPoolResults_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong queryPool, jlong firstQuery, jlong queryCount, jlong dataSize, jlong pData, jlong stride, jlong flags, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetQueryPoolResults f = (PFN_vkGetQueryPoolResults)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkQueryPool*)jniLongToPointer(queryPool)),
        *((uint32_t*)jniLongToPointer(firstQuery)),
        *((uint32_t*)jniLongToPointer(queryCount)),
        *((size_t*)jniLongToPointer(dataSize)),
        ((void*)jniLongToPointer(pData)),
        *((VkDeviceSize*)jniLongToPointer(stride)),
        *((VkQueryResultFlags*)jniLongToPointer(flags))
    );
}


jint Java_cz_mg_vulkan_PFNvkGetQueryPoolResults_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong queryPool, jint firstQuery, jint queryCount, jlong dataSize, jlong pData, jlong stride, jint flags){
    (void)env;
    (void)jc;
    PFN_vkGetQueryPoolResults f = (PFN_vkGetQueryPoolResults)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkQueryPool*)jniLongToPointer(queryPool)),
        (uint32_t)firstQuery,
        (uint32_t)queryCount,
        (size_t)dataSize,
        ((void*)jniLongToPointer(pData)),
        (VkDeviceSize)stride,
        (VkQueryResultFlags)flags
    );
}

